mlir/lib/Dialect/MIOpen/Generator/Conv2dGenerator.cpp

```diff
@@ -149,8 +149,16 @@ LogicalResult Conv2dGenerator::parseConvConfig(const char *arguments) {
       }
     };
 
-    // arch settings
-    strToStr("arch", config.arch);
+    // arch settings, use only the arch but not features
+    std::string archName;
+    strToStr("arch", archName);
+    std::size_t firstSeperatorLoc = archName.find(':');
+    if (firstSeperatorLoc == std::string::npos) {
+      config.arch = archName;
```

Comment:
Do we instead want to fail here? The string, if I understand correctly, will always have a `:` in it, so, if we don't see one, we should kick the whole parse out for violating the expected format.

Reply:
No we don't want to fail without a `:`. Pre-ROCm 4.1, the archName string will not include `:` or features. This is to make sure the client API call is backward compatible as MIOpen does.

Reply:
Good to know